test/document.test.js

```diff
@@ -11649,6 +11649,51 @@ describe('document', function() {
         assert.equal(res.counter, 2);
       });
 
+      it('calls setters on the value passed to `$inc()` (gh-13158)', async function() {
+        const called = [];
+        const Test2 = db.model('Test2', Schema({
+          counter: {
+            type: Number,
+            set: v => { called.push(v); return v.toFixed(2); }
+          }
+        }));
+        const doc = await Test2.create({ counter: 2 });
+        assert.deepEqual(called, [2]);
+
+        doc.$inc('counter', 1.14159);
+        assert.deepEqual(called, [2, 3.14159]);
+        assert.equal(doc.counter, 3.14);
+        await doc.save();
+
+        const res = await Test2.findById(doc);
+        assert.equal(res.counter, 3.14);
+      });
+
+      it('sets value even if setter fails (gh-13158)', async function() {
```

Comment:
i think this title now needs to be changed, because from my understanding it does not actually change the value anymore on fail